Desktop/Tempest/TempestServer.cs

```diff
@@ -205,7 +205,7 @@ protected virtual void OnConnectionMade (object sender, ConnectionMadeEventArgs
 				return;
 
 			lock (this.connections)
-				this.connections.Add (e.Connection, ExecutionMode.ConnectionOrder);
+				this.connections [e.Connection] = ExecutionMode.ConnectionOrder;
```

Comment:
Yeah, but I don't have any idea of what the larger bug could be. Any pointers? Here's the full trace, for the record:

```
System.ArgumentException: An element with the same key already exists in the dictionary.
at System.Collections.Generic.Dictionary`2[Tempest.IConnection,Tempest.ExecutionMode].Add (IConnection key, ExecutionMode value) [0x00000] in <filename unknown>:0 
  at Tempest.TempestServer.OnConnectionMade (System.Object sender, Tempest.ConnectionMadeEventArgs e) [0x00000] in <filename unknown>:0 
  at Tempest.Providers.Network.NetworkConnectionProvider.OnConnectionMade (Tempest.ConnectionMadeEventArgs e) [0x00000] in <filename unknown>:0 
  at Tempest.Providers.Network.NetworkConnectionProvider.Connect (Tempest.Providers.Network.NetworkServerConnection connection) [0x00000] in <filename unknown>:0 
  at Tempest.Providers.Network.NetworkServerConnection.OnTempestMessageReceived (Tempest.MessageEventArgs e) [0x00000] in <filename unknown>:0 
  at Tempest.Providers.Network.NetworkConnection.ReliableReceiveCompleted (System.Object sender, System.Net.Sockets.SocketAsyncEventArgs e) [0x00000] in <filename unknown>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.OnCompleted (System.Net.Sockets.SocketAsyncEventArgs e) [0x00000] in <filename unknown>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.ReceiveCallback (IAsyncResult ares) [0x00000] in <filename unknown>:0 
  at System.Net.Sockets.SocketAsyncEventArgs.DispatcherCB (IAsyncResult ares) [0x00000] in <filename unknown>:0 
```